### PR TITLE
julia-mono 0.022 → 0.034

### PR DESCRIPTION
**Pull request by toastal**
###### Motivation for this change

Upgrade to latest: https://github.com/cormullion/juliamono/releases/tag/v0.034
Some of the documentation changed as well like the website.

closes #104477 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tomberek** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114627` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>julia-mono</li>
</ul>
</details>

File: pkgs/data/fonts/julia-mono/default.nix [hunk 1]
```diff
@@ -14,9 +15,12 @@ in fetchzip {
 
   meta = {
     description = "A monospaced font for scientific and technical computing";
+    longDescription = ''
+      JuliaMono is a monospaced typeface designed for use in text editing environments that require a wide range of specialist and technical Unicode characters. It was intended as a fun experiment to be presented at the 2020 JuliaCon conference in Lisbon, Portugal (which of course didn’t physically happen in Lisbon, but online).
+    '';
     maintainers = with lib.maintainers; [ suhr ];
     platforms = with lib.platforms; all;
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/julia-mono/default.nix:22]: with lib should be moved to meta.

Comment (tomberek) [reply pkgs/data/fonts/julia-mono/default.nix:22]: Oops. New pr?

Comment (SuperSandro2000) [reply pkgs/data/fonts/julia-mono/default.nix:22]: More like a little cleanup. Going to put this into my clean collection.

**Pull request merged**